CMakeLists.txt

```diff
@@ -0,0 +1,428 @@
+cmake_minimum_required(VERSION 3.15)
+
+# Absolute path to FreeRTOS config file directory
+set(FREERTOS_CONFIG_FILE_DIRECTORY "" CACHE STRING "Absolute path to the directory with FreeRTOSConfig.h")
+
+# Heap number or absolute path to custom heap implementation provided by user
+set(FREERTOS_HEAP "4" CACHE STRING "FreeRTOS heap model number. 1 .. 5. Or absolute path to custom heap source file")
```

Comment:
Actually, there is no problem to have
```cmake
set(FREERTOS_HEAP "4" CACHE STRING "FreeRTOS heap model number. 1 .. 5. Or absolute path to custom heap source file")
```

Since it's `CACHE STRING`, setting any value to `FREERTOS_HEAP` overrides its default value. User can set its custom value to `FREERTOS_HEAP` from top-level CMake file (`set(FREERTOS_HEAP "5" CACHE STRING "")`) or command line (`-DFREERTOS_HEAP=5`)